plugins/vbe/README.md

```diff
@@ -0,0 +1,324 @@
+   is currently supported.
+* area-start - start offset in bytes of the firmware area within the storage
+* area-size - size in bytes of the firmware area within the storage
+* skip-offset - offset to preserve at the start of the firmware area. This
+   means that the first part of the image is ignored, with just the latter part
+   being written. For example, if this is 0x200 then the first 512 bytes of the
+   image (which must be present in the image) are skipped and the bytes after
+   that are written to the store offset.
+* part-uuid - the UUID of the partition containing the fwupd state. This is not
+   used by fwupd at present but may allow the bootloader to check the fwupd
+   state on boot
+* part-id - the partition number containing the fwupd state. This is not
+   used by fwupd at present but may allow the bootloader to check the fwupd
+   state on boot
+
+State is typically stored in /var/local/lib/fwupd/vbe/simple.dtb in device tree
+format:
+
+```devicetree
+ / {
+     compatible = "vbe";
+     vbe-driver = "fwupd,vbe-simple";
+     last-update {
+         finish-time = <0x00000000 0x62a7429e>;
+         cur-version = "1.2.4";
+         status = "completed";
+     };
+```
+
+The fields are as follows:
+
+* compatible - the compatible string for VBE
+* vbe-driver - the actual VBE method being used
+* last-update - information about the last update
+* finish-time - the unix time of the last update, i.e. seconds since 1970, as a
+    64-bit value (0 if no previous update)
+* cur-version - the current installed version, missing if there is none or it is
+    not known
+* status - status of the update. The only supported value is 'completed'
+
+## Adding new VBE methods
+
+New VBE methods can be added in the plugin file `fu-plugin-vbe.c` in the
+`driver_list` array. See the documentation there for how to do this.
+
+## Finding out more
+
+You can use the U-Boot mailing list or `u-boot` IRC on `libera.chat` to ask
+questions specific to VBE and firmware.
+
+## Sending patches
+
+Use the [fwupd developer site](https://fwupd.org/lvfs/docs/developers) to find
+information about downloading the code, submitting bugs/feature requests and
+sending patches.
+
+## Vendor ID Security
+
+This does not update USB devices and thus requires no vendor ID set.
+
+## External Interface Access
+
+This plugin requires access to system firmware, e.g. via a file or an eMMC
+device.
+
+## Documentation
+
+The following documents may help in understanding VBE:
+
+* [VBE](https://docs.google.com/document/d/e/2PACX-1vQjXLPWMIyVktaTMf8edHZYDrEvMYD_iNzIj1FgPmKF37fpglAC47Tt5cvPBC5fvTdoK-GA5Zv1wifo/pub)
+* [VBE Bootflows](https://docs.google.com/document/d/e/2PACX-1vQF7g5vjaoDzRDPWmbYOoOaNQsaPYDaCT0I2LZRNFIT0lD1hqANOcBmMicM-MACFQIjvE_VN9cSvCzy/pub)
+* [VBE Firmware update](https://docs.google.com/document/d/e/2PACX-1vTnlIL17vVbl6TVoTHWYMED0bme7oHHNk-g5VGxblbPiKIdGDALE1HKId8Go5f0g1eziLsv4h9bocbk/pub)
```

Comment:
Neither of these 2 URLs open for me.

Reply:
I fixed the first one

The second one works OK for me. Can you try again please?